src/main/java/com/keniu/services/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package com.keniu.services;
+
+import com.keniu.dto.CreateCartItemRequestDto;
+import com.keniu.dto.ShoppingCartDto;
+import com.keniu.dto.UpdateCarItemRequestDto;
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.mappers.ShoppingCartMapper;
+import com.keniu.models.Book;
+import com.keniu.models.CartItem;
+import com.keniu.models.ShoppingCart;
+import com.keniu.repositories.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    @Transactional
+    public ShoppingCart save(ShoppingCart shoppingCart) {
+        return shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addCartItem(Long userId,
+            CreateCartItemRequestDto createCartItemRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        CartItem cartItem = new CartItem();
+        cartItem.setShoppingCart(shoppingCart);
+        Book book = new Book();
+        book.setId(createCartItemRequestDto.getBookId());
+        cartItem.setBook(book);
+        cartItem.setQuantity(createCartItemRequestDto.getQuantity());
+        shoppingCart.getCartItems().add(cartItem);
+        shoppingCartRepository.save(shoppingCart);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto update(Long userId, Long id,
+            UpdateCarItemRequestDto updateCarItemRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        CartItem cartItem = getCartItemById(shoppingCart, id);
```

Comment:
can you use `findByIdAndShoppingCartId` to find cartItem?

Reply:
For what?

Reply:
I don't like this idea

Reply:
If you would be so kind, could you tell me why?